src/vs/workbench/browser/layout.ts

```diff
@@ -975,17 +976,38 @@ export abstract class Layout extends Disposable implements IWorkbenchLayoutServi
 		this._register(delegate.onDidChangeNotificationsVisibility(visible => this._onDidChangeNotificationsVisibility.fire(visible)));
 	}
 
-	hasFocus(part: Parts): boolean {
+	hasFocus(part: Parts, viewContainerId?: string): boolean {
 		const activeElement = document.activeElement;
 		if (!activeElement) {
 			return false;
 		}
 
+		const activePanel = this.getFocusWebviewPart(part);
+		if (viewContainerId && viewContainerId === activePanel?.getId()) {
+			return true;
+		}
+
 		const container = this.getContainer(part);
 
 		return !!container && isAncestorUsingFlowTo(activeElement, container);
```

Comment:
Have you investigated why `isAncestorUsingFlowTo` doesn't work in this case? Ideally this part of the code shouldn't have to know anything webview specific

Reply:
> Have you investigated why `isAncestorUsingFlowTo` doesn't work in this case? Ideally this part of the code shouldn't have to know anything webview specific



> Have you investigated why `isAncestorUsingFlowTo` doesn't work in this case? Ideally this part of the code shouldn't have to know anything webview specific

I think what you said is right. Ideally, there should be no special treatment for it. I'll try to see if `activeElement` value can be corrected.

Reply:
> Have you investigated why `isAncestorUsingFlowTo` doesn't work in this case? Ideally this part of the code shouldn't have to know anything webview specific

hi @mjbvz `activeElement` looks like a read-only property, we don't seem to be able to actively modify the value for it.